src/main/java/ca/corefacility/bioinformatics/irida/database/changesets/AutomatedAnalysisToTemplate.java

```diff
@@ -0,0 +1,168 @@
+		}
+
+		//get the workflow information from the service
+		try {
+			sistrWorkflow = workflowsService.getDefaultWorkflowByType(BuiltInAnalysisTypes.SISTR_TYPING);
+		} catch (IridaWorkflowNotFoundException e) {
+			logger.warn("SISTR workflow not found.  Automated SISTR will not be converted to analysis templates.");
+			//Note this will definitely happen in the galaxy CI tests as only SNVPhyl and a test workflow are configured.
+		}
+
+		//update assemblies
+		if (assemblyWorkflow != null) {
+			logger.debug("Upadting automated assembly project settings");
+
+			//get the workflow identifiers
+			UUID assemblyId = assemblyWorkflow.getWorkflowIdentifier();
+			//get the default parameters
+			List<IridaWorkflowParameter> defaultAssemblyParams = assemblyWorkflow.getWorkflowDescription()
+					.getParameters();
+
+			//insert the assembly
+			insertWorkflow(jdbcTemplate, "Automated AssemblyAnnotation", true, assemblyId, "p.assemble_uploads=1",
+					defaultAssemblyParams);
+
+		}
+
+		//update SISTR
+		if (sistrWorkflow != null) {
+			logger.debug("Upadting automated SISTR project settings");
+			//get the workflow identifiers
+			UUID sistrId = sistrWorkflow.getWorkflowIdentifier();
+
+			//get the default params
+			List<IridaWorkflowParameter> defaultSistrParams = sistrWorkflow.getWorkflowDescription()
+					.getParameters();
+
+			//insert the sistr entries without metadata
+			insertWorkflow(jdbcTemplate, "Automated SISTR Typing", false, sistrId, "p.sistr_typing_uploads='AUTO'",
+					defaultSistrParams);
+			//insert the sistr entries with metadata
+			insertWorkflow(jdbcTemplate, "Automated SISTR Typing", true, sistrId,
+					"p.sistr_typing_uploads='AUTO_METADATA'", defaultSistrParams);
+		}
+
+		return new SqlStatement[0];
+	}
+
+	private void insertWorkflow(JdbcTemplate jdbcTemplate, String name, boolean updateSamples, UUID workflowId,
+			String where, List<IridaWorkflowParameter> params) {
+		/*
+		 * we're borrowing the 'automated' flag here to mark entries we need to add params to later.  at this point
+		 * there'll be nothing with a '1' in the automated flag.  we'll clear it later.
+		 */
+
+		int updateSampleBit = updateSamples ? 1 : 0;
+
+		String assemblyInsert =
+				"INSERT INTO analysis_submission (DTYPE, name, created_date, priority, update_samples, workflow_id, submitter, submitted_project_id, automated) select 'AnalysisSubmissionTemplate', '"
+						+ name + "', now(), 'LOW', " + updateSampleBit + ", '" + workflowId.toString()
+						+ "', 1, p.id, 1 FROM project p WHERE " + where;
+
+		int update = jdbcTemplate.update(assemblyInsert);
+
+		//if we added anything, add the params
+		if (update > 0) {
+
+			// Insert the default params for the analysis type
+			for (IridaWorkflowParameter p : params) {
+				String assemblyParamInsert =
+						"INSERT INTO analysis_submission_parameters (id, name, value) SELECT a.id, '" + p.getName()
```

Comment:
I believe the parameter name/values be escaped when inserting into the database. It could be possible they contain quotes and other special characters (I don't think they do, but just in case).

Reply:
Changed all the insertions to be paramaterized in 4d037c6.  It involved a couple more `SELECT` queries, but this will be a quick changeset so no big deal.